source/frontend/src/components/common/GenericModal.tsx

```diff
@@ -134,29 +139,65 @@ export const GenericModal = (props: Omit<BsModalProps, 'onHide'> & ModalProps) =
     }
   };
 
+  const getHeaderIcon = () => {
+    if (headerIcon) {
+      return <>{headerIcon}</>;
+    }
+
+    switch (className) {
+      case 'info':
+      case 'warning': {
+        return <FaExclamationCircle size={22} />;
+      }
+      case 'error': {
+        return <FaTimesCircle size={22} />;
+      }
+      default: {
+        return null;
+      }
+    }
+  };
+
   return (
-    <ModalContainer {...rest} show={showState} modalSize={modalSize} onHide={noop}>
+    <ModalContainer
+      {...rest}
+      show={showState}
+      modalSize={modalSize}
+      onHide={noop}
+      className={className}
+    >
       <Modal.Header closeButton={closeButton} onHide={close}>
```

Comment:
the flex is on this div, which means the direct children are the title and the close button. That means the header icon and the text end up being grandchildren and end up in a display: block element within the title. That probably means you are going to end up with some wonky behaviour if the text in the title overlaps the length of the modal.